### DISCUSSION TITLE
Whats the right way to do this:

### DISCUSSION THREAD

**Discussion by IndianBoy42** [folke/flash.nvim#89] [Q&A] [answered]
```lua
  vim.go.operatorfunc = "v:lua.__remote_op_opfunc"
  vim.api.nvim_feedkeys("g@", "n", false)
      require("flash").remote(opts)
```
Assume `_G.__remote_op_opfunc` is set to some lua function.

I want to start operator-pending mode (with `g@`) and then immediately activate flash remote with some custom config. The way its written flash seems to not be activated at all. If I put the flash call inside of `vim.schedule` then its almost like `g@` gets run twice, I have to do another motion immediately

 If I use `feedkeys("g@r", "n", false)` instead, with no jump afterwards then the operator is called correctly.

**Comment by folke** [comment]
Just got home. Will see what's not working here

**Comment by folke** [comment, marked as answer]
I used `vim.api.nvim_feedkeys` to cancel the original op and used `vim.api.nvim_input` to schedule the new one.
For some reason, this led to the old one not being cancelled properly.
Shold be fixed now.

You can use the below:
```lua
_G.__remote_op_opfunc = function(motion)
  vim.notify("hi " .. motion)
end

vim.go.operatorfunc = "v:lua.__remote_op_opfunc"
vim.api.nvim_input("g@")

vim.schedule(function()
  require("flash").jump({
    remote_op = {
      restore = true,
      motion = false,
    },
  })
end)
```

Let me know if you'd still have issues!

**Reply by IndianBoy42** [reply]
Thanks for the help! Its nice to know I'm not the only one confused by nvim_feedkeys, nvim_input and vim.cmd.normal, and when each should be used.
